diff --git ql/src/java/org/apache/hadoop/hive/ql/optimizer/MapJoinProcessor.java ql/src/java/org/apache/hadoop/hive/ql/optimizer/MapJoinProcessor.java
index 275a31f52f..51f848cabf 100644
--- ql/src/java/org/apache/hadoop/hive/ql/optimizer/MapJoinProcessor.java
+++ ql/src/java/org/apache/hadoop/hive/ql/optimizer/MapJoinProcessor.java
@@ -231,9 +231,8 @@ public static void genLocalWorkForMapJoin(MapredWork newWork, MapJoinOperator ne
       newWork.getMapWork().setBaseSrc(null);
       newWork.getMapWork().setMapAliases(null);
     } catch (Exception e) {
-      e.printStackTrace();
       throw new SemanticException("Failed to generate new mapJoin operator " +
-          "by exception : " + e.getMessage());
+          "by exception : " + e);
     }
   }
 
